native/core/src/parquet/parquet_support.rs

```diff
@@ -171,7 +178,36 @@ fn cast_array(
                     .with_timezone(Arc::clone(tz)),
             ))
         }
-        _ => Ok(cast_with_options(&array, to_type, &PARQUET_OPTIONS)?),
+        // If Arrow cast supports the cast, delegate the cast to Arrow
+        _ if can_cast_types(from_type, to_type) => {
+            Ok(cast_with_options(&array, to_type, &PARQUET_OPTIONS)?)
+        }
+        _ => Ok(array),
+    }
+}
+
+fn cast_list_to_list(
+    array: &ListArray,
+    from_type: &DataType,
+    to_type: &DataType,
+    parquet_options: &SparkParquetOptions,
+) -> DataFusionResult<ArrayRef> {
+    match (from_type, to_type) {
+        (DataType::List(_), DataType::List(to_inner_type)) => {
```

Comment:
I think you could just pass `to_inner_type` directly into this method to avoid matching on `List` again here?

Reply:
Also `from_type` doesn't need to be passed in?

Reply:
simplified